src/rules/html/warn-on-inline.js

```diff
@@ -2,23 +2,21 @@ import { VALIDATION_WARNING } from 'const';
 import * as messages from 'messages';
 
 
-export function warnOnInline($, filename) {
-  return new Promise((resolve) => {
-    const linterMessages = [];
-    $('script').each((i, element) => {
-      if ($(element).attr('src') === undefined &&
-        ($(element).attr('type') === undefined ||
-        $(element).attr('type') === 'text/javascript')) {
-        linterMessages.push(
-          Object.assign({}, messages.INLINE_SCRIPT, {
-            /* This could occur in any HTML file, so let's make it
-             * a warning in case they've included any other file.
-             */
-            type: VALIDATION_WARNING,
-            file: filename,
-          }));
-      }
-    });
-    resolve(linterMessages);
+export async function warnOnInline($, filename) {
+  const linterMessages = [];
+  $('script').each((i, element) => {
+    if ($(element).attr('src') === undefined &&
+      ($(element).attr('type') === undefined ||
+      $(element).attr('type') === 'text/javascript')) {
+      linterMessages.push(
+        Object.assign({}, messages.INLINE_SCRIPT, {
+          /* This could occur in any HTML file, so let's make it
+            * a warning in case they've included any other file.
+            */
+          type: VALIDATION_WARNING,
+          file: filename,
+        }));
+    }
   });
+  return linterMessages;
```

Comment:
Nit (optional, but nice) add an empty line between the `});` and the `return linterMessage`.

Reply:
@rpl all changes have been made